### ISSUE TITLE
Too many open issues

### ISSUE THREAD

**Issue by patrick96** [jaagr/polybar#802]
This project has grown quite a bit over the last year. And also recently the number of new issues picked up a lot. While it is great that more people use the project and get involved, it quickly can get overwhelming getting a bunch of new issues every day, many of which would take quite a lot of time to address. Having this many issues open can also lead to an issue getting lost next to all the other open ones.

With this thread, I want to start a conversation and get feedback on how we can reduce the numbers of issues opened as well as the average time it takes to process issues (especially more trivial ones).
This will give us maintainers more time to address more critical issues.

I will start listing some ideas that I had but I also want to hear from the other maintainers @jaagr, @NBonaparte and @skystrife and of course you, the people using polybar and reporting issues, what you think we can improve, so that people don't necessarily have to open an issue and can solve their problem on their own, as well as, if they do open an issue it can be processed by us more quickly.

## Contributing Guidelines
Creating a `Contributing.md` file will prompt a user creating a new issue to review the contributing guidelines, for more info click [here](https://help.github.com/articles/setting-guidelines-for-repository-contributors/).
This file could contain instructions on how to structure an issue depending on what kind of issue it is (bug, build error, feature request, etc.). It could also tell the user what we expect they have already done before actually opening the issue. Some of which are
- Having looked at the wiki
- Using the latest stable release or the latest commit
- Tried to solve it with a `custom/script` module
- ...

## Issue Templates
This is very close to the point before. Here we could have the user confirm that they read the `Contributing.md` file and other things (like read the Wiki, etc.). See the [Atom Issue Template](https://github.com/atom/atom/blob/master/ISSUE_TEMPLATE.md) as an example.

## Preventing Question Issues
Many tickets are of the form "How can I do X?" and "Is it possible to do Y?", often such questions can be resolved by responding with a link to the wiki or a simple "This is not possible.". We want to keep those types of questions to a minimum by enabling the user to find the information themselves. Github allows us to create a `Support.md` file and it will also prompt the user to read that before opening an issue, as described [here](https://help.github.com/articles/adding-support-resources-to-your-project/).

## Adding an FAQ
This would be another way to decrease question style issues.

## Closing Old Issues
If the original author stops responding to requests, the issue is often forgotten about and remains open. This happens frequently if the author does not tell you if a proposed solution to their issue worked. In the past, once we stumbled upon such an issue, we just closed it, if we felt enough time had passed. I think we should formalize, how and when we close such issues.
I propose the following rules:
- If the author was able to reproduce the issue and fix it on their machine, they can close the issue once the PR with the fix is merged or immediately, if the fix doesn't require any code changes to the project.
- Else close the issue after 30 days. Maybe consider using [stalebot](https://github.com/probot/stale) for this.

## Use the invalid label
If an issue comes in where the author clearly has not read or followed the Contributing Guidelines, simply makr it `invalid` with a note telling them to try again.

What do you think about this so far?

I will draft some PRs to add contributing guidelines, the issue template and maybe even start the FAQ.

**Comment by NBonaparte** [comment]
I agree with all of this.
Issue templates would be very helpful, as there have been multiple occasions where we have asked for more information.
We should definitely have automatic issue closing. I recall a while back there were a lot of old issues that were resolved which we had to close manually.

**Comment by jaagr** [comment]
Great initiative. I also agree with all of this. I don't know how many times I've asked for more information regarding an issue such as config, version, etc. Having a bit more restrictive policy to issue reports (list of requirements for example) would solve part of it. I do however think that there's a fine line between useful restrictions and restrictions that make the reporter think twice about actually reporting - which wouldn't be good.

There might be other bots (a part from stalebot that, from what I can tell, only handles inactivity) that could help with commenting on missing details, auto closing, etc. Worth looking into.
